src/state/connection-mixin.ts

```diff
@@ -275,6 +276,23 @@ export const connectionMixin = <T extends Constructor<HassBaseEl>>(
       subscribeFrontendUserData(conn, "core", (userData) =>
         this._updateHass({ userData })
       );
+
+      const pingBackend = async () => {
+        if (this.hass?.connected) {
+          promiseTimeout(5000, this.hass?.connection.ping()).catch(() => {
+            if (!this.hass?.connected) {
+              return;
+            }
+
+            // eslint-disable-next-line no-console
+            console.log("Websocket died, forcing reconnect...");
+            this.hass?.connection.reconnect(true);
+          });
+        }
+        setTimeout(pingBackend, 5000);
```

Comment:
Use `setInterval` instead

Reply:
We should stop the interval on disconnected and start it again on reconnected.

To make absolutely sure it is not running twice, just stop it before starting a new one :-)